iceoryx_posh/test/moduletests/test_posh_runtime_single_process.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#
+#include "iceoryx_posh/internal/roudi_environment/roudi_environment.hpp"
+#include "iceoryx_posh/runtime/posh_runtime_single_process.hpp"
+
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::runtime;
+using namespace iox::roudi;
+
+namespace iox
+{
+namespace test
+{
+/// @brief Test goal: This test suit verifies class posh_runtime_single_process
+
+class PoshRuntimeSingleProcess_test : public Test
+{
+  public:
+    PoshRuntimeSingleProcess_test()
+    {
+    }
+
+    virtual ~PoshRuntimeSingleProcess_test()
+    {
+    }
+
+    virtual void SetUp(){};
+
+    virtual void TearDown(){};
+};
+
+TEST_F(PoshRuntimeSingleProcess_test, ConstructorPoshRuntimeSingleProcessIsSuccess)
+{
+    iox::RouDiConfig_t defaultRouDiConfig = iox::RouDiConfig_t().setDefaults();
+    IceOryxRouDiComponents roudiComponents(defaultRouDiConfig);
+
+    RouDi roudi(roudiComponents.m_rouDiMemoryManager,
+                roudiComponents.m_portManager,
+                RouDi::RoudiStartupParameters{iox::roudi::MonitoringMode::OFF, false});
+
+    const ProcessName_t m_runtimeName{"App"};
+
+    EXPECT_NO_FATAL_FAILURE({ PoshRuntimeSingleProcess m_runtimeSingleProcess(m_runtimeName); });
+}
+
+TEST_F(PoshRuntimeSingleProcess_test, ConstructorPoshRuntimeSingleProcessMultipleProcessIsFound)
+{
+    RouDiEnvironment m_roudiEnv{iox::RouDiConfig_t().setDefaults()};
+
+    const ProcessName_t m_runtimeName{"App"};
+
+    iox::cxx::optional<iox::Error> detectedError;
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&detectedError](const iox::Error error, const std::function<void()>, const iox::ErrorLevel errorLevel) {
+            detectedError.emplace(error);
+            EXPECT_THAT(errorLevel, Eq(iox::ErrorLevel::FATAL));
+        });
+
+    PoshRuntimeSingleProcess m_runtimeSingleProcess(m_runtimeName);
```

Comment:
This is not expected for me that the `PoshRuntimeSingleProcess` is instantiated only once and throws an error. This should come up if you create the objects twice or more. When i try to add an additional testcase, this will not work because the same error is thrown. Looks like that the sut (this should be named here) is not properly cleaned up in the previous test.

This smells for me like that the ctor/dtor of `PoshRuntimeSingleProcess` has a problem.
@elfenpiff can it be that the `defaultRuntimeFactory` is not properly set in the PoshRuntime?

Reply:
@dkroenke Everything is quiet alright here. The `PoshRuntime` is created once as a Singleton and then lives as long the process is running. The same goes for the `PoshRuntimeSingleProcess` where the only difference is that you run RouDi in your process to use him for InterThreadCommunication in one process.

Reply:
@elfenpiff Ok i thought that it is possible to create a new `PoshRuntimeSingleProcess` with a new name after destroying the previous one, but i guess that's only works when the underlying PoshRuntime with the new name is recreated.